legion/engine/rendering/data/particle_system_base.cpp

```diff
@@ -0,0 +1,64 @@
+#include <rendering/data/particle_system_base.hpp>
+#include <rendering/components/particle_emitter.hpp>
+#include <rendering/components/renderable.hpp>
+
+
+namespace legion::rendering
+{
+    ecs::EcsRegistry* ParticleSystemBase::m_registry;
+
+    void ParticleSystemBase::createParticle(ecs::component_handle<particle> particleHandle, transform transformHandle) const
+    {
+        ecs::entity_handle particularParticle = particleHandle.entity;
+
+        //Handle model and material assigning.
+        particularParticle.add_component<renderable>({m_particleModel, m_particleMaterial});
+    }
+
+    void ParticleSystemBase::cleanUpParticle(ecs::component_handle<particle> particleHandle, ecs::component_handle<particle_emitter> emitterHandle) const
+    {
+        //Read emitter
+        particle_emitter emitter = emitterHandle.read();
+        const auto particle = std::find(emitter.livingParticles.begin(), emitter.livingParticles.end(), particleHandle);
+        if (particle != emitter.livingParticles.end())
+        {
+            ecs::entity_handle particularParticle = *particle;
+            //Remove from living
+            const auto iterator = std::remove(emitter.livingParticles.begin(), emitter.livingParticles.end(), particularParticle);
+            emitter.livingParticles.erase(iterator);
+            //Add to dead
+            emitter.deadParticles.emplace_back(particularParticle);
+            //Remove renderable to stop them from being rendered
+            particularParticle.remove_component<renderable>();
+        }
+        //Write to emitter
+        emitterHandle.write(emitter);
```

Comment:
shouldn't this have happened behind the scenes, and shouldn't the cleanup particle only remove the renderable component?